src/index.js

```diff
@@ -17,65 +18,38 @@ const DEFAULT_OPTIONS = {
 };
 
 module.exports = function pluginGraphQLDocGenerator(context, opts) {
-  // Merge defaults with user-defined options.
-  const config = { ...DEFAULT_OPTIONS, ...opts };
+  // Merge defaults with user-defined options in config file.
+  pluginConfigs.push({ ...DEFAULT_OPTIONS, ...opts });
 
   return {
     name: "docusaurus-graphql-doc-generator",
 
     extendCli(cli) {
       cli
         .command("graphql-to-doc")
-        .option("-s, --schema <schema>", "Schema location", config.schema)
-        .option(
-          "-r, --root <rootPath>",
-          "Root folder for doc generation",
-          config.rootPath,
-        )
-        .option(
-          "-b, --base <baseURL>",
-          "Base URL to be used by Docusaurus",
-          config.baseURL,
-        )
-        .option(
-          "-l, --link <linkRoot>",
-          "Root for links in documentation",
-          config.linkRoot,
-        )
+        .option("-s, --schema <schema>", "Schema location")
+        .option("-r, --root <rootPath>", "Root folder for doc generation")
+        .option("-b, --base <baseURL>", "Base URL to be used by Docusaurus")
+        .option("-l, --link <linkRoot>", "Root for links in documentation")
         .option(
           "-h, --homepage <homepage>",
           "File location for doc landing page",
-          config.homepage,
         )
         .option("-f, --force", "Force document generation")
-        .option("-d, --diff <diffMethod>", "Set diff method", config.diffMethod)
-        .option(
-          "-t, --tmp <tmpDir>",
-          "Set temp dir for schema diff",
-          config.tmpDir,
-        )
+        .option("-d, --diff <diffMethod>", "Set diff method")
+        .option("-t, --tmp <tmpDir>", "Set temp dir for schema diff")
         .option(
           "-gbd, --groupByDirective <@directive(field|=fallback)>",
           "Group Documentation By Directive",
-          config.groupByDirective,
         )
         .option("--pretty", "Prettify generated files")
         .description("Generate GraphQL Schema Documentation")
         .action(async (options) => {
-          await generateDocFromSchema({
-            baseURL: options.base,
-            schemaLocation: options.schema,
-            outputDir: path.join(options.root, options.base),
-            linkRoot: options.link,
-            homepageLocation: options.homepage,
-            diffMethod: options.force ? "FORCE" : options.diff,
-            tmpDir: options.tmp,
-            loaders: config.loaders,
-            groupByDirective:
-              GroupInfo.parseOption(options.groupByDirective) ||
-              config.groupByDirective,
-            prettify: options.pretty || config.pretty,
-          });
+          for (const config of pluginConfigs) {
+            await generateDocFromSchema(
+              mergeConfigWithCLIOptions(config, options),
+            );
+          }
```

Comment:
Let's create a `run()` function to simplify the code:

```js
\\\....
        .action(async (options) => {
          await run(pluginConfigs, options);
        });
    },
  };
};

async function run(pluginConfigs, cliOptions) {
  for (const config of pluginConfigs) {
    await generateDocFromSchema(buildConfig(config, cliOptions));
  }
}
```

Reply:
Not needed since we chose the "one command instance per config" approach.